### ISSUE TITLE
Dovecot 2.3.1 fails to start

### ISSUE THREAD

**Issue by noorul** [Homebrew/homebrew-core#26157]
- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh

To help us debug your issue please explain:
- What you were trying to do (and why)

I upgraded dovecot from 2.3.0.1 to 2.3.1

- What happened (include command output)

I see the following in dovecot log 

Fatal: service(stats) Group doesn't exist: dovecot (See service stats { unix_listener /usr/local/var/run/dovecot/stats-writer { group } } setting)
Apr 02 06:52:07 master: Fatal: service(stats) Group doesn't exist: dovecot (See service stats { unix_listener /usr/local/var/run/dovecot/stats-writer { group } } setting)
Fatal: service(stats) Group doesn't exist: dovecot (See service stats { unix_listener /usr/local/var/run/dovecot/stats-writer { group } } setting)
Apr 04 10:11:43 master: Fatal: service(stats) Group doesn't exist: dovecot (See service stats { unix_listener /usr/local/var/run/dovecot/stats-writer { group } } setting)

In dovecot master conf file I have the following declared

default_login_user = _dovenull
default_internal_user = _dovecot

I am not sure why is it still looking for dovecot group. I don't see any change to the config file syntax in release notes https://dovecot.org/pipermail/dovecot-news/2018-March/000375.html

Going back to 2.3.0.1 fixes the problem.

- What you expected to happen

Expected dovecot server to start

- Step-by-step reproduction instructions (by running `brew install` commands)

brew upgrade dovecot

**Comment by fxcoudert** [comment]
You have deleted an important part of the template:

> - [ ] ran `brew update` and can still reproduce the problem?
> - [ ] ran `brew doctor`, fixed all issues and can still reproduce the problem?
> - [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
> - [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

We need this information so we can help you.

**Comment by noorul** [comment]
- [ ] ran brew update and can still reproduce the problem?
- [ ] ran brew doctor, fixed all issues and can still reproduce the problem?
- [ ] ran brew gist-logs <formula> (where <formula> is the name of the formula that failed) and included the output link?
- [ ] if brew gist-logs didn't work: ran brew config and brew doctor and included their output with your issue?

**Comment by noorul** [comment]
$ --> brew config
    HOMEBREW_VERSION: 1.5.14
    ORIGIN: https://github.com/Homebrew/brew
    HEAD: 7fd6210127f088b6ee8708a1d7f4ec2df3fc5bb4
    Last commit: 2 days ago
    Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
    Core tap HEAD: bd45eddcf6eeff673d9eab77c32a1d1fc2fd08af
    Core tap last commit: 5 hours ago
    HOMEBREW_PREFIX: /usr/local
    CPU: quad-core 64-bit kabylake
    Homebrew Ruby: 2.3.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
    Clang: 9.1 build 902
    Git: 2.17.0 => /usr/local/bin/git
    Curl: 7.54.0 => /usr/bin/curl
    Java: 1.8.0_162
    macOS: 10.13.4-x86_64
    CLT: 9.3.0.0.1.1521514116
    Xcode: N/A
    XQuartz: N/A

     Nooruls-MacBook-Pro in ~
    $ --> brew doctor
    Your system is ready to brew.

     Nooruls-MacBook-Pro in ~
    $ -->

**Comment by drx777** [comment]
I have the same issue after updating dovecot today

```
Fatal: service(stats) Group doesn't exist: dovecot (See service stats { unix_listener /usr/local/var/run/dovecot/stats-writer { group } } setting)
```

```
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have unlinked kegs in your Cellar
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  docker-machine
  php
  gst-plugins-base
  samba
  lesstif
  gstreamer
```

```
$ brew config
HOMEBREW_VERSION: 1.5.14
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 7fd6210127f088b6ee8708a1d7f4ec2df3fc5bb4
Last commit: 4 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: beffa1b97ff0c2fdd37a228fed6496e437134e42
Core tap last commit: 2 hours ago
HOMEBREW_PREFIX: /usr/local
CPU: octa-core 64-bit kabylake
Homebrew Ruby: 2.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby
Clang: 9.1 build 902
Git: 2.17.0 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 10, 9.0.4, 1.8.0_91, 1.7.0_11, 1.7.0_06
macOS: 10.13.4-x86_64
CLT: 9.3.0.0.1.1521514116
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11
```

**Comment by drx777** [comment]
This is what I got from the dovecot IRC channel:
> the 2.2 stats plugin is called old_stats in 2.3

> release notes for 2.3.0 mention just new stats plugin but there's more info here regarding the change: https://wiki2.dovecot.org/Statistics

I think I may give it a try to remove the old configuration and replace it with a current one.

**Comment by ilovezfs** [comment]
Closing since this works as designed according to the current caveats and Homebrew configuration file handling, but PRs are welcome to propose modifications to the behavior.

**Comment by drx777** [comment]
I had to reinstall 2.2.33.2 via
```
brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/8e04df3ae828b0f5a9654ce99c4a3688ca641eae/Formula/dovecot.rb
```
and it worked again.

I couldn't make 2.3.1 work, although I added the dovecot group. I even removed all var and log data and started with a new configuration. I could login via doveadm auth login <user>, but not with my mail client.

I hope there will be a better solution or more information on what exactly is to be done in order to make dovecot 2.3.1 work here.
